src/box/key_def.h

```diff
@@ -307,6 +307,9 @@ typedef struct key_def box_key_def_t;
 enum {
 	BOX_KEY_PART_DEF_IS_NULLABLE = 1 << 0,
 	BOX_KEY_PART_DEF_EXCLUDE_NULL = 1 << 1,
+	BOX_KEY_PART_DEF_SORT_ORDER = 1 << 2,
+	BOX_KEY_PART_DEF_SORT_ORDER_ASC = 0,
```

Comment:
I don't understand why we need this empty flag. Why not add one flag `BOX_KEY_PART_DEF_IS_REVERSE`? Come to think, is there any point in having a enum in `key_part` for describing the sort order? Wouldn't a boolean flag (`is_reverse`) suffice? Can there be any other sort order except "ascending" and "descending"?

Reply:
The flag is de-facto boolean, but it is represented in the structure as part of the `flags` field to consume one bit instead of one byte. This economy is due to the frozen size of the public structure and a wish to have the ability to add flags/fields for a long time without breaking the binary layout.

Regarding the name of the flag: it looks natural to name it with words similar to ones from the Lua API, hence 'sort order'.

I agree that the idea with two additional constants for ascending and descending orders is controversial. One should be OR'ed, another should be AND'ed, we anyway need to know the values and the abstraction is leaking.

How about one constant for the mask: BOX_KEY_PART_DEF_SORT_ORDER_DESC? It is quite long, sure, but that's mostly due to namespacing.

BOX_KEY_PART_DEF_IS_DESCENDING and BOX_KEY_PART_DEF_IS_DESC are OK too.

All the options have pros and cons, so I'll accept any of them.

Reply:
Dropped the `BOX_KEY_PART_DEF_SORT_ORDER_ASC` and called the mask `BOX_KEY_PART_DEF_SORT_ORDER_DESC`.